### PR TITLE
Append cached replacement node when a node is removed in KBucket.

### PR DESCRIPTION
**Pull request by zsaladin**
### What was wrong?
I found Replacement nodes are not used in KBucket.

### How was it fixed?
When the number of nodes in KBucket decreases, one of the replacement nodes is added.

#### Cute Animal Picture

put a cute animal picture link inside the parentheses

### CODE REVIEW

**Comment by zsaladin** [comment]
Added a test for this.

**Comment by zsaladin** [comment]
I have a question.
If a replacement node of a KBucket is passed as a parameter in `remove_node` then what will happen?

Removing process will be ignored in current codes.
If the replacement node has to be removed from KBucket then I will fix it.
Please give me your opinion. @pipermerriam 

Thank you.

File: tests/p2p/test_kademlia.py [hunk 1]
```diff
@@ -116,6 +116,30 @@ def test_kbucket_add():
...
+    replacement_nodes = [random_node() for _ in range(replacement_count)]
+    for replacement_node in replacement_nodes:
+        bucket.add(replacement_node)
+    assert len(bucket) == bucket.k
+
+    for node in nodes:
+        bucket.remove_node(node)
+    assert len(bucket) == replacement_count
+
+    for replacement_node in replacement_nodes:
```
Comment (pipermerriam) [inline tests/p2p/test_kademlia.py]: Maybe add one more set of assertions before removing these to check that each node in `replacement_nodes` is in the bucket, as well as asserting they are *not* in the bucket prior to removing all of the original nodes.

**Comment by pipermerriam** [comment]
I opened an issue to track the issue about `remove_node` being called with a node in the list of replacement nodes https://github.com/ethereum/trinity/issues/187

File: tests/p2p/test_kademlia.py (hunk 1 above)
Comment (zsaladin) [reply tests/p2p/test_kademlia.py]: I modified test codes to clear how it works.

**Comment by pipermerriam** [comment]
@zsaladin thanks for finding and fixing this.  much appreciated.

**Pull request merged**